src/cluster_legacy.c

```diff
@@ -5843,7 +6002,16 @@ int clusterNodePending(clusterNode *node) {
     return node->flags & (CLUSTER_NODE_NOADDR | CLUSTER_NODE_HANDSHAKE);
 }
 
-char *clusterNodeIp(clusterNode *node) {
+/* Returns the IP of the node as seen by the given client, or by the cluster node if c is NULL. */
+char *clusterNodeIp(clusterNode *node, client *c) {
+    if (c == NULL) {
+        return node->ip;
+    }
+    if (isClientConnIpV6(c)) {
+        if (sdslen(node->client_ipv6) != 0) return node->client_ipv6;
+    } else {
```

Comment:
do we not want to fall back to "cluster-announce-ip" per https://github.com/valkey-io/valkey/pull/736/files#diff-dd9a6c95849b36490b9bae44e9f0d3dbcd7a93080cf2d5aee6d6033586ab8fa4R1804?

Reply:
Yeah, `cluster-announce-ip` is stored in `node->ip`, return below.